src/components/calcite-slider/calcite-slider.scss

```diff
@@ -1,63 +1,88 @@
-$thumb-size: 28px;
-$handle-size: 14px;
-$thumb-padding: ($thumb-size - $handle-size) / 2;
-$track-height: 2px;
-$tick-height: 4px;
+:host([scale="s"]) .container {
+  --calcite-slider-handle-size: 10px;
+  --calcite-slider-handle-extension-height: 6.5px;
+  --calcite-slider-container-font-size: var(--calcite-font-size--3);
+  .handle__label,
+  .tick__label {
+    @apply leading-3;
+  }
+}
 
-@mixin histogramEndcaps() {
-  .tick__label--min,
-  .tick__label--max {
-    margin: 6px -3px;
-    font-weight: 300;
-    color: var(--calcite-ui-text-3);
+:host([scale="m"]) .container {
+  --calcite-slider-handle-size: 14px;
+  --calcite-slider-handle-extension-height: 8px;
+  --calcite-slider-container-font-size: var(--calcite-font-size--2);
+  .handle__label,
+  .tick__label {
+    @apply leading-4;
   }
 }
 
+:host([scale="l"]) .container {
+  --calcite-slider-handle-size: 16px;
+  --calcite-slider-handle-extension-height: 10.5px;
+  --calcite-slider-container-font-size: var(--calcite-font-size--1);
+  .handle__label,
+  .tick__label {
+    @apply leading-4;
+  }
+}
+
+.handle__label,
+.tick__label {
+  @apply font-medium text-color-2;
+  font-size: var(--calcite-slider-container-font-size);
+}
+
 :host {
   @apply block;
 }
 
+:host(:not([has-histogram])) {
+  @apply w-full;
+}
+
 .container {
   @apply block relative;
-  padding: $handle-size / 2 0;
-  margin: $handle-size / 2 theme("margin.2");
+  padding: calc(var(--calcite-slider-handle-size) / 2) 0;
+  margin: calc(var(--calcite-slider-handle-size) / 2) theme("margin.2");
```

Comment:
To-do: set container margin all the way around instead of just top/bottom, after pr #3004 is merged.